addons/binding/org.openhab.binding.onebusaway/src/main/java/org/openhab/binding/onebusaway/internal/handler/StopHandler.java

```diff
@@ -0,0 +1,273 @@
+        if (config == null) {
+            logger.debug("Initialization of OneBusAway bridge failed!");
+            return;
+        }
+
+        // Do the rest of the work asynchronously because it can take a while.
+        scheduler.submit(new Runnable() {
+            @Override
+            public void run() {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_PENDING, "Initializing...");
+                httpClient = new HttpClient();
+                try {
+                    httpClient.start();
+                } catch (Exception e) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, e.getMessage());
+                    httpClient = null;
+                    return;
+                }
+                gson = new Gson();
+
+                if (!fetchAndUpdateStopData()) {
+                    return;
+                }
+
+                scheduleJob();
+            }
+        });
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void dispose() {
+        if (pollingJob != null && !pollingJob.isCancelled()) {
+            pollingJob.cancel(true);
+            pollingJob = null;
+        }
+        if (httpClient != null) {
+            httpClient.destroy();
+            httpClient = null;
+        }
+    }
+
+    /**
+     * Registers the listener to receive updates about arrival and departure times for its route.
+     *
+     * @param listener
+     * @return true if successful.
+     */
+    protected boolean registerRouteDataListener(RouteDataListener listener) {
+        if (listener == null) {
+            throw new IllegalArgumentException("It makes no sense to register a null listener!");
+        }
+        boolean added = routeDataListeners.add(listener);
+        if (added) {
+            String routeId = listener.getRouteId();
+            List<ObaStopArrivalResponse.ArrivalAndDeparture> copiedRouteData;
+            synchronized (routeData) {
+                copiedRouteData = Lists.newArrayList(routeData.get(routeId));
+            }
+            Collections.sort(copiedRouteData);
+            listener.onNewRouteData(routeDataLastUpdateMs, copiedRouteData);
+        }
+        return added;
+    }
+
+    /**
+     * Unregisters the listener so it no longer receives updates about arrival and departure times for its route.
+     *
+     * @param listener
+     * @return true if successful.
+     */
+    protected boolean unregisterRouteDataListener(RouteDataListener listener) {
+        return routeDataListeners.remove(listener);
+    }
+
+    /**
+     * Forced an update to be scheduled immediately, and reschedules our normal task to run again on our normal interval
+     * starting now.
+     */
+    protected void forceUpdate() {
+        if (pollingJob != null) {
+            pollingJob.cancel(true);
+            pollingJob = null;
+        }
+        scheduler.execute(pollingRunnable);
```

Comment:
I would reduce this method to simply `scheduler.execute(pollingRunnable);`.
This method should do a fetch "out of the row", I would not change the existing polling job in here - this isn't done in any other binding either and it makes the overall logic much simpler and better understandable.